closed/adds/jdk/src/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -100,34 +101,43 @@ protected void engineInit(Key key, SecureRandom random)
                 ("Key must be an instance of PrivateKey");
         }
         /* attempt to translate the key if it is not an ECKey */
-        this.privateKey = (ECPrivateKeyImpl) ECKeyFactory.toECKey(key);
-        this.publicKey = null;
+        ECKey ecKey = ECKeyFactory.toECKey(key);
+        if (ecKey instanceof ECPrivateKeyImpl) {
+            this.privateKey = (ECPrivateKeyImpl) ecKey;
+            this.publicKey = null;
```

Comment:
Clearing `publicKey` happened unconditionally before; perhaps this assignment should occur before the `instanceof` test?
```
            this.publicKey = null;
```

Reply:
If `ecKey` is not an instance of `ECPrivateKeyImpl`, we use the java implementation, so `publicKey` in `NativeECDHKeyAgreement` is not used. And the `publicKey` in `ECDHKeyAgreement` will be cleared in `this.javaImplementation.engineInit(key, random)`.